include/forcing/ForcingsEngineDataProvider.hpp

```diff
@@ -105,13 +108,7 @@ struct ForcingsEngineDataProvider : public DataProvider<DataType, SelectionType>
     using clock_type = std::chrono::system_clock;
 
     ~ForcingsEngineDataProvider() override = default;
-    void finalize() override
-    {
-        if (bmi_ != nullptr) {
-            bmi_->Finalize();
-            bmi_ = nullptr;
-        }
-    }
+    void finalize() override = default;
```

Comment:
This doesn't build with `= default`, because it's not a constructor or destructor, but just an ordinary named member function.

I would have said "Since it's now a no-op, we can just delete it", but I instead think that it should probably be
```suggestion
    void finalize() override
    {
        bmi_.reset();
    }
```
I would like the eventual call to `clear()` from `ForcingsEngineStorage::finalize()` to actually release the object reference.